src/pagos_client.rs

```diff
@@ -0,0 +1,127 @@
+}
+
+impl PagosApiClient {
+    pub fn new(base_url: String, api_key: String) -> PagosClientResult<Self> {
+        let client = reqwest::Client::builder()
+            .build()
+            .map_err(PagosClientError::NetworkError)?;
+        Ok(Self {
+            client,
+            base_url,
+            api_key,
+        })
+    }
+
+    pub async fn get_pan_details(
+        &self,
+        bin_number: &str,
+    ) -> PagosClientResult<PagosPanDetailsResponse> {
+        let url = format!("{}/bins?enhanced=true&bin={}", self.base_url, bin_number);
+
+        let mut headers = header::HeaderMap::new();
+        headers.insert(
+            "x-api-key",
+            header::HeaderValue::from_str(&self.api_key).map_err(|_| {
+                error_stack::Report::new(PagosClientError::ConfigError(
+                    "Invalid Pagos API key format".to_string(),
+                ))
+            })?,
+        );
+        headers.insert(
+            header::ACCEPT,
+            header::HeaderValue::from_static("application/json"),
+        );
+
+        let response = self
+            .client
+            .get(&url)
+            .headers(headers)
+            .send()
+            .await
+            .map_err(PagosClientError::NetworkError)?;
+
+        if response.status().is_success() {
+            let body_text = response
+                .text()
+                .await
+                .map_err(PagosClientError::NetworkError)?;
+            logger::debug!(pagos_api_response_body = %body_text, "Pagos API successful response");
+            Ok(serde_json::from_str::<PagosPanDetailsResponse>(&body_text)
+                .map_err(PagosClientError::DeserializationError)?)
+        } else {
+            let status = response.status();
+            let body = response
+                .text()
+                .await
+                .unwrap_or_else(|_| "Failed to read error body".to_string());
+            logger::error!(pagos_api_error_status = %status, pagos_api_error_body = %body, "Pagos API error");
+            Err(error_stack::Report::new(PagosClientError::ApiError {
+                status,
+                body,
+            }))
+        }
+    }
+}
+
+pub async fn fetch_pan_details_internal(
+    app_state: &TenantAppState,
+    card_isin: &str,
+) -> CustomResult<PagosPanDetailsResponse, ApiError> {
+    let pagos_config = app_state
+        .config
+        .pagos_api
+        .as_ref()
+        .ok_or(error_stack::Report::new(
+            ApiError::ParsingError("Pagos API configuration not found"),
+        ))
+        .attach_printable(
+            "Pagos API configuration not found for tenant when fetching PAN details internally",
+        )?;
+
+    let pagos_client = PagosApiClient::new(
+        pagos_config.base_url.clone(),
+        pagos_config.api_key.clone().expose().clone(),
+    )
+    .change_context(ApiError::UnknownError)
+    .attach_printable("Failed to initialize Pagos API client during internal fetch")?;
```

Comment:
```suggestion
    let pagos_config = app_state
        .pagos_client
    .change_context(ApiError::UnknownError)
    .attach_printable("Failed to initialize Pagos API client during internal fetch")?;
```

create pagos client while creating the tenantAppState itself